gamemodes/terrortown/gamemode/client/cl_targetid.lua

```diff
@@ -240,37 +226,12 @@ function GM:HUDDrawTargetID()
 		DrawPropSpecLabels(client)
 	end
 
-	local startpos = client:EyePos()
-	local endpos = client:GetAimVector()
-
-	endpos:Mul(MAX_TRACE_LENGTH)
-	endpos:Add(startpos)
-
 	local ent, unchangedEnt, distance
+	local startpos = client:EyePos()
+	local direction = client:GetAimVector()
+	local filter = client:GetObserverMode() == OBS_MODE_IN_EYE and {client, client:GetObserverTarget()} or client
 
-	-- if the user is looking at a traitor button, it should always be handled with priority
-	if TBHUD.focus_but and IsValid(TBHUD.focus_but.ent) and (TBHUD.focus_but.access or TBHUD.focus_but.admin) and TBHUD.focus_stick >= CurTime() then
-		ent = TBHUD.focus_but.ent
-
-		distance = startpos:Distance(ent:GetPos())
-	else
-		local trace = util.TraceLine({
-			start = startpos,
-			endpos = endpos,
-			mask = MASK_SHOT,
-			filter = client:GetObserverMode() == OBS_MODE_IN_EYE and {client, client:GetObserverTarget()} or client
-		})
-
-		-- this is the entity the player is looking at right now
-		ent = trace.Entity
-
-		distance = trace.StartPos:Distance(trace.HitPos)
-	end
-
-	-- if a vehicle, we identify the driver instead
-	if IsValid(ent) and IsValid(ent:GetNWEntity("ttt_driver", nil)) then
-		ent = ent:GetNWEntity("ttt_driver", nil)
-	end
+	ent, distance = targetid.FindEntityAlongView(startpos, direction, filter)
 
 	-- only add onscreen infos when the entity isn't the local player
 	if ent == client then return end
```

Comment:
I know, this line is by me. But shouldn't it be `if ent == filter then return end`. The filter variable itself wouldn't work here since it returns a table. But it would solve problems with spectators probably. Do you know what I mean?

Reply:
Yeah, I think I understand. But it shouldn't matter, as the findEntity-function filters those entities already out.
So theoretically it would suffice to delete that line. No entity that got filtered should get past the targetid-function.